java/spring/security/injection/tainted-html-string.yaml

```diff
@@ -32,46 +32,74 @@ rules:
     interfile: true
   mode: taint
   pattern-sources:
-  - patterns:
+  - label: INPUT
+    patterns:
     - pattern-either:
       - pattern-inside: |
-          $METHODNAME(..., @$REQ(...) $TYPE 
-          $SOURCE,...) {
+          $METHODNAME(..., @$REQ(...) $TYPE $SOURCE,...) {
             ...
           }
       - pattern-inside: |
-          $METHODNAME(..., @$REQ $TYPE 
-          $SOURCE,...) {
+          $METHODNAME(..., @$REQ $TYPE $SOURCE,...) {
             ...
           }
     - metavariable-regex:
         metavariable: $REQ
         regex: (RequestBody|PathVariable|RequestParam)
-    - pattern: $SOURCE
+    - focus-metavariable: $SOURCE
+  - label: CONCAT
+    by-side-effect: true
+    requires: INPUT
+    patterns:
+    - pattern-either:
+      - pattern: |
+          "$HTMLSTR" + ...
+      - pattern: |
+          "$HTMLSTR".concat(...)
+      - patterns:
+        - pattern-inside: |
+            StringBuilder $SB = new StringBuilder("$HTMLSTR");
```

Comment:
Can these be typed MVars?

Reply:
wanted to specify that the creation of it needs to have a "HTMLSTR" with something that looks like an html tag

Reply:
ya it does help with FPs 👍